Src/Support/Google.Apis.Auth/OAuth2/ImpersonatedCredential.cs

```diff
@@ -35,65 +36,102 @@ namespace Google.Apis.Auth.OAuth2
+
+            /// <summary>
+            /// Gets the scopes to request during the authorization grant. May be null or empty.
+            /// </summary>
+            public IEnumerable<string> Scopes { get; set; }
+
+            /// <summary>
+            /// Gets or sets for how long the delegated credential should be valid.
+            /// Defaults to 1 hour or 3600 seconds.
+            /// </summary>
+            public TimeSpan Lifetime { get; set; }
 
             /// <summary>Constructs a new initializer.</summary>
-            /// <param name="sourceCredential">The source credential used to acquire the impersonated credentials.</param>
-            /// <param name="options">The impersonation options.</param>
-            public Initializer(GoogleCredential sourceCredential, ImpersonationOptions options)
-                : base(string.Format(GoogleAuthConsts.IamAccessTokenEndpointFormatString, options.TargetPrincipal))
-            { 
-                SourceCredential = sourceCredential;
-                Options = options;
+            /// <param name="targetPrincipal">The principal that will be impersonated. Must not be null.</param>
+            public Initializer(string targetPrincipal)
+                : base(string.Format(GoogleAuthConsts.IamAccessTokenEndpointFormatString, targetPrincipal.ThrowIfNull(nameof(targetPrincipal))))
+            {
+                TargetPrincipal = targetPrincipal;
+                Lifetime = TimeSpan.FromHours(1);
             }
 
-            internal Initializer(ImpersonatedCredential other) : base(other.TokenServerUrl)
+            internal Initializer(ImpersonatedCredential other) : base(other)
             {
-                SourceCredential = other.SourceCredential;
-                Options = other.Options;
+                TargetPrincipal = other.TargetPrincipal;
+                DelegateAccounts = other.DelegateAccounts;
+                Scopes = other.Scopes;
+                Lifetime = TimeSpan.FromHours(1);
             }
         }
 
         /// <summary>
         /// Gets the source credential used to acquire the impersonated credentials.
         /// </summary>
-        public GoogleCredential SourceCredential { get; }
+        public GoogleCredential SourceCredential => HttpClientInitializers.OfType<GoogleCredential>().Single();
+
+        /// <summary>
+        /// Gets the service account to impersonate.
+        /// </summary>
+        public string TargetPrincipal { get; }
+
+        /// <summary>
+        /// Gets the chained list of delegate service accounts. May be empty.
+        /// </summary>
+        public IEnumerable<string> DelegateAccounts { get; }
+
+        /// <summary>
+        /// Gets the scopes to request during the authorization grant. May be empty.
+        /// </summary>
+        public IEnumerable<string> Scopes { get; }
 
         /// <summary>
-        /// Gets the impersonation options.
+        /// Gets for how long the delegated credential should be valid.
```

Comment:
I'm not a big fan of using "gets" in general, but let's keep doing so for consistency. However, I'd reword this to something like "Gets the lifetime of the delegated credential. This is how long the delegated credential should be valid." If we can clarify what that means, that would be useful too - presumably it's "from the time of the first request made with this credential"?